modules/tinymce/src/themes/silver/test/ts/webdriver/throbber/ThrobberTabbingTest.ts

```diff
@@ -0,0 +1,112 @@
+    Insert.before(target, inputBefore);
+    Insert.append(SugarBody.body(), inputAfter);
+  });
+
+  after(() => {
+    Arr.each([ '#beforeInput', '#afterInput' ], (selector) => {
+      Remove.remove(SelectorFind.descendant(SugarBody.body(), selector).getOrDie());
+    });
+  });
+
+  const pAssertFocus = (label: string, selector: string) =>
+    FocusTools.pTryOnSelector(label, SugarDocument.getDocument(), selector);
+
+  const pAssertThrobberFocus = () =>
+    pAssertFocus('Throbber has focus', 'div.tox-throbber__busy-spinner');
+
+  const pAssertEditorFocus = (editor: Editor) => () =>
+    pAssertFocus('Editor has focus', editor.inline ? 'div.mce-edit-focus' : 'iframe.tox-edit-area__iframe');
+
+  const pAssertInputFocus = (before: boolean) => () =>
+    pAssertFocus('Input has focus', before ? '#beforeInput' : '#afterInput');
+
+  const pPressTab = (selector: string, shiftKey?: boolean) =>
+    RealKeys.pSendKeysOn(selector, [ RealKeys.combo({ shiftKey }, '\u0009') ] );
+
+  const pEnableThrobber = async (editor: Editor, pAssertFocus: () => Promise<SugarElement>) => {
+    editor.setProgressState(true);
+    await UiFinder.pWaitForVisible('Wait for throbber to show', SugarBody.body(), '.tox-throbber');
+    await pAssertFocus();
+  };
+
+  const pDisableThrobber = async (editor: Editor, pAssertFocus: () => Promise<SugarElement>) => {
+    editor.setProgressState(false);
+    await UiFinder.pWaitForHidden('Wait for throbber to hide', SugarBody.body(), '.tox-throbber');
+    await pAssertFocus();
+  };
+
+  it('TINY-7373: should be able to tab into editor if throbber is disabled', async () => {
+    const editor = hook.editor();
+    FocusTools.setFocus(SugarBody.body(), '#beforeInput');
+    await pPressTab('#beforeInput');
+    await pAssertEditorFocus(editor)();
+    assert.isTrue(editor.hasFocus());
+  });
+
+  it('TINY-7373: should take focus if editor is tabbed into and throbber is enabled', async () => {
+    const editor = hook.editor();
+    FocusTools.setFocus(SugarBody.body(), '#beforeInput');
+    await pEnableThrobber(editor, pAssertInputFocus(true));
+
+    // Make sure throbber gets focus if the editor is tabbed into
+    await pPressTab('#beforeInput');
+    await pAssertThrobberFocus();
+
+    await pDisableThrobber(editor, pAssertEditorFocus(editor));
+    assert.isTrue(editor.hasFocus());
+  });
+
+  it('TINY-7373: should be able to Tab out of the throbber if it has focus', async () => {
+    const editor = hook.editor();
+    FocusTools.setFocus(SugarBody.body(), '#beforeInput');
+    await pEnableThrobber(editor, pAssertInputFocus(true));
+
+    await pPressTab('#beforeInput');
+    await pAssertThrobberFocus();
+
+    await pPressTab('div.tox-throbber__busy-spinner');
+    await pAssertInputFocus(false)();
+
+    await pDisableThrobber(editor, pAssertInputFocus(false));
+    assert.isFalse(editor.hasFocus());
+  });
+
+  // Automated test doesn't work on Safari but works manually
+  if (!Env.browser.isSafari()) {
```

Comment:
This isn't how we skip tests, it should be done inside the function instead. eg:
```
it('TINY-7373: should be able to Shift+Tab out of the throbber if it has focus', async function () {
  if (Env.browser.isSafari()) {
    this.skip();
  }
```

- [ ] Fix how this test is being skipped on Safari